splunk_add_on_ucc_framework/ucc_ui_lib/src/main/webapp/components/BaseFormView.jsx

```diff
@@ -117,56 +132,113 @@ class BaseFormView extends Component {
-        let validator = new Validator(this.entities);
+        const validator = new Validator(this.entities);
         let error = validator.doValidation(datadict);
         if (error) {
             this.setErrorFieldMsg(error.errorField, error.errorMsg);
         } else if (this.options && this.options.saveValidator) {
             error = SaveValidator(this.options.saveValidator, datadict);
             if (error) {
                 this.setErrorMsg(error.errorMsg);
+                
             }
         }
 
-        const saveSuccess = !error;
+        if (error && this.hook && typeof this.hook.onSaveFail === 'function') {
+            this.hook.onSaveFail();
+            this.props.handleFormSubmit(false,false);
+        }
+        else if(error){
+            this.props.handleFormSubmit(false,false);
+        }
+
+        if(!error){
+            const params = new URLSearchParams();
 
-        const returnValue = {
-            result: saveSuccess,
-            data: datadict,
-        };
+            Object.keys(datadict).forEach( (key) => {
+                if(datadict[key]){
+                params.append(key, datadict[key] );
+                }
+            });
 
-        if (saveSuccess) {
-            if (this.hook && typeof this.hook.onSaveSuccess === 'function') {
-                this.hook.onSaveSuccess();
+            if(this.props.mode ===MODE_EDIT){
+                params.delete("name");
             }
-            return returnValue;
-        }
 
-        if (this.hook && typeof this.hook.onSaveFail === 'function') {
-            this.hook.onSaveFail();
+            axiosCallWrapper({
+                serviceName: this.endpoint,
+                params,
+                customHeaders: { 'Content-Type': 'application/x-www-form-urlencoded' },
+                method: 'post',
+                handleError: false
+            }).catch((err) => {
+                const errorSubmitMsg= this.parseErrorMsg(err?.response?.data?.messages[0]?.text);
+                this.setState({ErrorMsg:errorSubmitMsg});
+                if (this.hook && typeof this.hook.onSaveFail === 'function') {
+                    this.hook.onSaveFail();
+                }
+                this.props.handleFormSubmit(false,false);
+                return Promise.reject(err);
+            }).then((response) => {
+                
+                const val = response?.data?.entry[0];
+                const tmpObj ={};
+                
+                tmpObj[val.name] = {
+                    ...val.content,
+                    id: val.id,
+                    name: val.name,
+                    serviceName: this.props.serviceName
+                };
+
+                this.context.setRowData( update(this.context.rowData,{[this.props.serviceName]: {$merge : tmpObj}}))
+                this.props.handleFormSubmit(false,true);
+                
+            });
         }
-        return returnValue;
     };
+
+    parseErrorMsg = (msg) => {
```

Comment:
Keep single copy of this function instead of duplicating it in multiple files.

Reply:
done